scripts/sas_nmap_baseline.sh

```
@@ -0,0 +1,208 @@
+    "$(json_escape "$ts")" "$(json_escape "$RUN_ID")" "$(json_escape "$stage")" "$(json_escape "$level")" "$(json_escape "$msg")" "$(json_escape "$data")" >> "$EVENT_LOG"
+  printf '[%s][%s] %s\n' "$stage" "$level" "$msg" | tee -a "$TRACE_LOG"
+}
+
+capture() {
+  local label="$1" out="$2"; shift 2
+  { echo "# $label"; echo "# command: $*"; echo "# captured_at: $(now_iso)"; echo; } > "$out"
+  if [[ "$DRY_RUN" -eq 1 ]]; then echo "DRY RUN" >> "$out"; echo "# exit_code: 0" >> "$out"; return 0; fi
+  set +e; "$@" >> "$out" 2>&1; code=$?; set -e
+  echo; echo "# exit_code: $code"
+  } >> "$out"
+}
```

Shellcheck (0.11.0)

[error] Parsing stopped here. Is this keyword correctly matched up?

(SC1089)